Runner.php

```diff
@@ -15,32 +15,65 @@ public function __construct($config, $server)
 
     public function queueDirectory($dir, $prefix = '')
     {
+        if(isset($this->config['include_paths'])) {
+            $this->queueWithIncludePaths();
+        } else {
+            $this->queuePaths($dir, $prefix);
+        }
+
+        $this->server->process_work(false);
+    }
+
+    public function queueWithIncludePaths() {
+        foreach ($this->config['include_paths'] as $f) {
+            if ($f !== '.' and $f !== '..') {
+
+                if (is_dir("/code$f")) {
+                    $this->queuePaths("/code$f", "$f/");
+                    continue;
+                }
+
+                if (isset($this->config['config']['file_extensions'])) {
+                    $this->filterByExtension($f);
+                } else {
+                    $this->server->addwork(array("/code/$f"));
+                }
+            }
+
+        }
+    }
+
+    public function queuePaths($dir, $prefix = '') {
         $dir = rtrim($dir, '\\/');
 
         foreach (scandir($dir) as $f) {
-            if (in_array("$prefix$f", $this->config["exclude_paths"])) {
+            if (!isset($this->config['include_paths']) && in_array("$prefix$f", $this->config["exclude_paths"])) {
```

Comment:
Sorry for the delayed reply. To be honest I am not sure if I got your idea right. How would the `include_paths` be handled if I pass an empty array? It's worth mentioning that at the moment, an engine decides to expand `include_paths` when this config value is not empty and the way CLI works after the recent changes is that it translates exclude_paths to include_paths.
As a result `include_paths` are the ones always getting expanded.